### PR TITLE
Phone number input

### PR DESCRIPTION
**Pull request by oleksandr-samoilenko-ew**
add intl_phone_number_input to otp email

[Screen_recording_20240418_130021.webm](https://github.com/extrawest/extrawest_ui_kit_mobile/assets/68539527/679781df-726c-4156-9522-09f86689bf25)

### CODE REVIEW

File: lib/layouts/otp_verification/otp_enter_phone.dart [hunk 1]
```diff
@@ -14,46 +14,138 @@ class OTPEnterPhoneNumber extends StatelessWidget {
...
-            hintText: phoneNumberHintText,
+          InternationalPhoneNumberInput(
+            onInputChanged: (PhoneNumber number) {
+              widget.onInputChanged?.call(number);
+            },
+            onInputValidated: (bool value) {
+              widget.onInputValidated?.call(value);
+            },
+            hintText: widget.phoneNumberHintText,
+            selectorConfig: const SelectorConfig(
```
Comment (andriy-gulak-ew) [inline lib/layouts/otp_verification/otp_enter_phone.dart:103]: allow custom selectorConfig

File: lib/layouts/otp_verification/otp_enter_phone.dart [hunk 2]
```diff
@@ -14,46 +14,138 @@ class OTPEnterPhoneNumber extends StatelessWidget {
...
+              labelText: widget.searchLabelText,
+              errorText: widget.searchErrorText,
+              helperText: widget.searchHelperText,
+              labelStyle: widget.searchLabelStyle,
+              // filled: true,
+              contentPadding: widget.searchContentPadding,
+            ),
+            formatInput: true,
+            keyboardType: const TextInputType.numberWithOptions(signed: true, decimal: true),
+            inputBorder: const OutlineInputBorder(),
```
Comment (andriy-gulak-ew) [inline lib/layouts/otp_verification/otp_enter_phone.dart:140]: allow custom object

File: lib/layouts/otp_verification/otp_enter_phone.dart [hunk 3]
```diff
@@ -14,46 +14,138 @@ class OTPEnterPhoneNumber extends StatelessWidget {
...
+            },
+            hintText: widget.phoneNumberHintText,
+            selectorConfig: const SelectorConfig(
+              selectorType: PhoneInputSelectorType.BOTTOM_SHEET,
+              useBottomSheetSafeArea: true,
+              leadingPadding: 20,
+              setSelectorButtonAsPrefixIcon: true,
+            ),
+            ignoreBlank: false,
+            autoValidateMode: AutovalidateMode.disabled,
```
Comment (andriy-gulak-ew) [inline lib/layouts/otp_verification/otp_enter_phone.dart:110]: custom

File: lib/layouts/otp_verification/otp_enter_phone.dart [hunk 4]
```diff
@@ -1,59 +1,126 @@
...
+                signed: true, decimal: true),
+            inputBorder: widget.customInputBorder,
+            onSaved: (PhoneNumber number) {
+              widget.onSaved?.call(number);
+            },
+            onSubmit: () {
+              widget.onSubmit?.call();
+            },
+            validator: (String? value) {
+              return widget.onValidate?.call(value ?? '');
```
Comment (andriy-gulak-ew) [inline lib/layouts/otp_verification/otp_enter_phone.dart:114]: try to simplify it like that: `validator: widget.onValidate`

File: lib/layouts/otp_verification/otp_enter_phone.dart [hunk 5]
```diff
@@ -1,59 +1,126 @@
...
+            onInputChanged: (PhoneNumber number) {
+              widget.onInputChanged?.call(number);
+            },
+            onInputValidated: (bool value) {
+              widget.onInputValidated?.call(value);
+            },
+            selectorConfig: widget.customSelectorConfig,
+            ignoreBlank: false,
+            autoValidateMode: widget.customAutoValidateMode,
+            selectorTextStyle: const TextStyle(color: Colors.black),
```
Comment (andriy-gulak-ew) [inline lib/layouts/otp_verification/otp_enter_phone.dart:98]: move it to param as well

File: lib/layouts/otp_verification/otp_enter_phone.dart (hunk 4 above)
Comment (oleksandr-samoilenko-ew) [reply lib/layouts/otp_verification/otp_enter_phone.dart:114]: cant figure out how to do that. We get   final String? Function(String?)? validator,  in the phone number input package, and it seems  String value is mandatory

**Review by andriy-gulak-ew — APPROVED** [review]

**Pull request merged**